src/ucp/wireup/select.c

```diff
@@ -1691,11 +1691,23 @@ ucp_wireup_add_rma_bw_lanes(const ucp_wireup_select_params_t *select_params,
 
         /* Add lanes that can access the memory by short operations */
         added_lanes = 0;
+        UCS_BITMAP_CLEAR(&tl_bitmap);
+
         for (mem_type = 0; mem_type < UCS_MEMORY_TYPE_LAST; mem_type++) {
+            UCP_CONTEXT_MEM_CAP_TLS(context, mem_type, reg_mem_types,
+                                    mem_type_tl_bitmap);
+            if (UCS_BITMAP_IS_ZERO_INPLACE(&mem_type_tl_bitmap)) {
+                continue;
+            }
```

Comment:
why is it needed, wouldn't ucp_wireup_add_bw_lanes return 0 lanes in this case anyway?

Reply:
No, for instance if for HOST memory we selected one IB and one tcp lane, then on the next round it will add more IB lanes for CUDA. This leads to asymmetrical lanes selection if GDR is enabled on one of the peers only

Reply:
right, so I'd think that just adding the UCP_TL_BITMAP_AND_NOT and UCS_BITMAP_OR_INPLACE below would be needed, and the check 'if UCS_BITMAP_IS_ZERO_INPLACE' can be removed.